### ISSUE TITLE
Create dataset loader for MiniMayoSRS Semantic Similarity Reference Standard

### ISSUE THREAD

**Issue by galtay** [bigscience-workshop/biomedical#162]
- **Name:** MiniMayoSRS Semantic Similarity Reference Standard
- **Description:** MiniMayoSRS is a subset of the MayoSRS and consists of 30 term pairs on which a higher inter-annotator agreement was achieved. The average correlation between physicians is 0.68. The average correlation between medical coders is 0.78.
- **Task:** Semantic Similarity
- **Paper:** https://pubmed.ncbi.nlm.nih.gov/16875881/
- **Data:** https://nlp.cs.vcu.edu/data.html#mayosrs
- **License:** ?
- **Motivation:**

**Comment by trishalaneeraj** [comment]
#self-assign

**Comment by jason-fries** [comment]
Hi @trishalaneeraj  can you let us know if you are still working on this so we can update our project board? Please just notify us the status **by Friday April 8**.  You can response to this comment or ping us on Slack or Discord. 

No worries if you are not finished but still intend to work on this!

**Comment by trishalaneeraj** [comment]
Hi @jason-fries, I'm still working on this

**Comment by debajyotidatta** [comment]
Hi @trishalaneeraj : This looks great! But in the source schema do you mind changing the values that are string to feature type float? It would be nice to have the float as the original source schema since that was intended.

**Comment by nomisto** [comment]
I am currently doing QA for this dataset and I ran into several issues:

1. Unit test fail because the data could not be downloaded (Would be fixed with #532)
2. The source data seems to have several flaws which result in wrong examples
    1. [Term Pairs](https://nlp.cs.vcu.edu/data/similarity-data/MiniMayoSRS.terms) and [physician judgments](https://nlp.cs.vcu.edu/data/similarity-data/MiniMayoSRS.physicians) have different sizes (30 vs 29)
    2. The indices of "Term Pairs" and "physician judgments" are not the same. f.e. the sample of `a["train"][28]` is `{'text_1': 'Varicose vein', 'text_2': 'Entire knee meniscus', 'label_physicians': 1.0, 'code_1_physicians': 'C0020473', 'code_2_physicians': 'C0027627', 'label_coders': 1.0, 'code_1_coders': 'C0020473', 'code_2_coders': 'C0027627'}`, however CUI C0020473 is Hyperlipidemia and C0027627 Tumor metastasis
3. There is no need to have  code_1_physicians and code_1_coders; they HAVE TO be the same (same goes for code_2)

**Comment by trishalaneeraj** [comment]
Hi @nomisto, did something change with the requirements of the tests? I did confirm that all tests passed on my end before submitting my PR. Here are the logs - 

`(bigscience-biomedical) ➜  biomedical git:(minimayosrs) python -m tests.test_bigbio biodatasets/minimayosrs/minimayosrs.py
INFO:__main__:args: Namespace(path='biodatasets/minimayosrs/minimayosrs.py', schema=None, subset_id=None, data_dir=None, use_auth_token=None)
INFO:__main__:self.PATH: biodatasets/minimayosrs/minimayosrs.py
INFO:__main__:self.SUBSET_ID: minimayosrs
INFO:__main__:self.SCHEMA: None
INFO:__main__:self.DATA_DIR: None
INFO:__main__:Checking for _SUPPORTED_TASKS ...
INFO:__main__:Found _SUPPORTED_TASKS=[<Tasks.SEMANTIC_SIMILARITY: 'STS'>]
INFO:__main__:_SUPPORTED_TASKS implies _MAPPED_SCHEMAS={'PAIRS'}
INFO:__main__:schemas_to_check: {'PAIRS'}
INFO:__main__:Checking load_dataset with config name minimayosrs_source
Downloading and preparing dataset minimayosrs_dataset/minimayosrs_source to /Users/trishalaneeraj/.cache/huggingface/datasets/minimayosrs_dataset/minimayosrs_source/1.0.0/e4c40f351a430ffb08e8c951a4b01d8019c114150e0af4b5a145b5cf31a37d15...
Downloading data files: 100%|█████████████████████████████████████████████████████| 3/3 [00:00<00:00, 1799.36it/s]
Extracting data files: 100%|███████████████████████████████████████████████████████| 3/3 [00:00<00:00, 285.82it/s]
Dataset minimayosrs_dataset downloaded and prepared to /Users/trishalaneeraj/.cache/huggingface/datasets/minimayosrs_dataset/minimayosrs_source/1.0.0/e4c40f351a430ffb08e8c951a4b01d8019c114150e0af4b5a145b5cf31a37d15. Subsequent calls will reuse this data.
100%|██████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 153.99it/s]
INFO:__main__:Checking load_dataset with config name minimayosrs_bigbio_pairs
Downloading and preparing dataset minimayosrs_dataset/minimayosrs_bigbio_pairs to /Users/trishalaneeraj/.cache/huggingface/datasets/minimayosrs_dataset/minimayosrs_bigbio_pairs/1.0.0/e4c40f351a430ffb08e8c951a4b01d8019c114150e0af4b5a145b5cf31a37d15...
Downloading data files: 100%|█████████████████████████████████████████████████████| 3/3 [00:00<00:00, 7332.70it/s]
Extracting data files: 100%|██████████████████████████████████████████████████████| 3/3 [00:00<00:00, 1281.09it/s]
Dataset minimayosrs_dataset downloaded and prepared to /Users/trishalaneeraj/.cache/huggingface/datasets/minimayosrs_dataset/minimayosrs_bigbio_pairs/1.0.0/e4c40f351a430ffb08e8c951a4b01d8019c114150e0af4b5a145b5cf31a37d15. Subsequent calls will reuse this data.
100%|██████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 461.22it/s]
INFO:__main__:Checking global ID uniqueness
INFO:__main__:Found 29 unique IDs
INFO:__main__:Gathering schema statistics
INFO:__main__:Gathering schema statistics
train
==========
id: 29
document_id: 29
text_1: 29
text_2: 29
label: 29

.
----------------------------------------------------------------------
Ran 1 test in 0.819s

OK`

I will definitely look into the issues you highlighted above as soon as possible. Thanks for bringing it to my attention.

**Comment by nomisto** [comment]
Hi @trishalaneeraj, 
Regarding 1.: It is weird that you could download the files. For me it threw an error 
```
File "...minimayosrs.py", line 161, in 
_generate_examples
    text_1, text_2 = row[0].split("<>")
ValueError: not enough values to unpack (expected 2, got 1)
```
Thats because for me the script downloads a html file rather than the resource. This is because if you haven't accept cookies it automatically redirects you to a webpage where you have to accept them. Maybe they introduced the "accept cookie" redirection inbetween or you could download the files somehow else. 
And regarding 2.: That is something the tests do not catch unfortunately.

However, I went ahead and fixed it in #540 just now (I found the dataset on another website https://conservancy.umn.edu/handle/11299/196265), you can take a look. Thanks again for your contribution!
